controller/reserves.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"go-project/model"
+	"io/ioutil"
+	"net/http"
+
+	"go-project/service"
+
+	"github.com/gorilla/mux"
+)
+
+type (
+	ReserveController struct {
+		reserveService *service.ReserveService
+	}
+
+	ReserveControllerImpl interface {
+		//CreateNewBooking(w http.ResponseWriter, r *http.Request)
+	}
+)
+
+func NewController(reserveService *service.ReserveService) *ReserveController {
+	return &ReserveController{
+		reserveService: reserveService,
+	}
+}
+
+func (rc *ReserveController) HomePage(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Welcome to the HomePage!")
+	fmt.Println("Endpoint Hit: homePage")
+}
+
+func (rc *ReserveController) CreateNewBooking(w http.ResponseWriter, r *http.Request) {
+	reqBody, _ := ioutil.ReadAll(r.Body)
+	var reserve model.Reserve
+	json.Unmarshal(reqBody, &reserve)
+	ctx := context.Background()
+	err := rc.reserveService.CreateNewBooking(ctx, reserve)
+	if err != nil {
+		json.NewEncoder(w).Encode(err)
+	}
+	json.NewEncoder(w).Encode("Success")
```

Comment:
Please add whitespace between code and handle all errors.
After a week, it will be very helpful to you and your team. It will also be easier to read and maintain.

like this:
```golang
  reqBody, err := ioutil.ReadAll(r.Body)
  if err != nil {
      // handle error from read here.
  }

  var reserve model.Reserve
  if err := json.Unmarshal(reqBody, &reserve); err != nil {
    // handle error from unmarshal here.
  }

  if err := rc.reserveService.CreateNewBooking(r.Context(), reserve); err != nil {
   // handle error from service here.
  }

  json.NewEncoder(w).Encode("Success")
```
and yeah, we doesn't use `context.Background()` outside `main` and `test` files.